### PR TITLE
Skeleton VerifiableCredentialStore API.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
WIP

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,16 @@ npm install
...
 ## Usage
 
-TBD
+```js
+import {getRemoteStorage} from 'bedrock-web-data-hub';
+import VerifiableCredentialStore from 'bedrock-web-vc-store';
+
+const remoteStorage = await getRemoteStorage({accountId: 'test'});
+
+const vcStore = new VerifiableCredentialStore({accountId: 'test', remoteStorage});
```
Comment (dmitrizagidulin) [inline README.md]: @dlongley Is this the right idea, in terms of creating a `VerifiableCredentialStore` instance?

Does the upstream code have to set up a `remoteStorage`, and pass it into the VC store?
Is the VC store dependent on a particular `accountId`?

Comment (dlongley) [reply README.md]: @dmitrizagidulin, I recommend upstream passes in a RemoteStorage API for now -- and you don't concern yourself with "accounts" at all. We'll see if that flies.

File: lib/VerifiableCredentialStore.js [hunk 2]
```diff
@@ -3,5 +3,161 @@
...
+      }
+    }
+    if(issuer) {
+      equals.push({issuer});
+    }
+    const results = await this.hub.find({equals});
+    return results.map(({content}) => content);
+  }
+
+  async _queryByExample({credentialQuery}) {
```
Comment (dlongley) [inline lib/VerifiableCredentialStore.js]: Move private functions to the bottom of the file/class

File: lib/VerifiableCredentialStore.js [hunk 3]
```diff
@@ -3,5 +3,161 @@
...
+  /**
+   * Finds the best matching verifiable credential for the given query
+   *
+   * @param query a VerifiablePresentation credential handler query
+   *   (e.g. right now, support `QueryByExample`).
+   * @param engine
+   *
+   * @return {Promise<Array>} List of matching VCs
+   */
+  async match({query, engine = credentials => credentials}) {
```
Comment (dlongley) [inline lib/VerifiableCredentialStore.js]: The default here makes no sense... should it be?

```suggestion
  async match({query, engine = credentialQuery => []}) {
```

File: test/web/10-VerifiableCredentialStore.test.js [hunk 4]
```diff
@@ -1,11 +1,158 @@
...
+    const vcStore = new VerifiableCredentialStore({hub});
+    
+    await vcStore.insert({credential: AlumniCredential});
+    const credential = await vcStore.get({id: AlumniCredential.id});
+
+    credential.should.be.an('object');
+    credential.should.deep.equal(AlumniCredential);
+  });
+
+
```
Comment (dlongley) [inline test/web/10-VerifiableCredentialStore.test.js]:
```suggestion
```

File: lib/VerifiableCredentialStore.js [hunk 5]
```diff
@@ -3,5 +3,161 @@
...
+    if(!credentialQuery) {
+      throw new Error('"credentialQuery" is needed to execute a QueryByExample.');
+    }
+    if(typeof credentialQuery !== 'object') {
+      throw new Error('"credentialQuery" must be an object or an array.');
+    }
+  
+    // normalize query to be an array
+    let query;  
+    if(credentialQuery instanceof Array) {
```
Comment (dlongley) [inline lib/VerifiableCredentialStore.js]:
```suggestion
    if(Array.isArray(credentialQuery)) {
```

File: lib/VerifiableCredentialStore.js [hunk 6]
```diff
@@ -3,5 +3,161 @@
...
+      query = credentialQuery;
+    } else {
+      query = [credentialQuery];
+    }
+
+    const _query = async ({example}) => {
+      const {type, trustedIssuer = []} = example;
+      // normalize trusted issuers to be an array
+      let trustedIssuers;
+      if(trustedIssuer instanceof Array) {
```
Comment (dlongley) [inline lib/VerifiableCredentialStore.js]:
```suggestion
      if(Array.isArray(trustedIssuer)) {
```

File: lib/VerifiableCredentialStore.js [hunk 7]
```diff
@@ -3,5 +3,161 @@
...
+   *
+   * @param {string} [type]
+   * @param {string} [issuer]
+   *
+   * @return {Promise<Array>} List of matching VCs
+   */
+  async find({type, issuer}) {
+    const equals = [];
+    if(type) {
+      if(type instanceof Array) {
```
Comment (dlongley) [inline lib/VerifiableCredentialStore.js]:
```suggestion
      if(Array.isArray(type)) {
```

**Pull request closed without merging**